test/helpers/reducers.ts

```diff
@@ -18,9 +18,8 @@ export interface Todo {
   id: number
   text: string
 }
-export type TodoAction = { type: 'ADD_TODO'; text: string } | AnyAction
 
-export function todos(state: Todo[] = [], action: TodoAction) {
+export function todos(state: Todo[] = [], action: rootActionTypes) {
```

Comment:
Now that the `action` is properly typed as `rootActionTypes`, the switch statement will narrow down the types to the appropriate one. This type of stuff is what TypeScript is all about!